iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -76,20 +82,22 @@ public override async Task SendEventAsync(Message message, CancellationToken can
                 if (Logging.IsEnabled)
                     Logging.Enter(this, message, cancellationToken, nameof(SendEventAsync));
 
+                using var operationCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _cancelPendingOperations.Token);
+
                 await _internalRetryPolicy
                     .RunWithRetryAsync(
                         async () =>
                         {
```

Comment:
Should we consider checking if the cancellation token is signaled and throw before retry attempt?

Reply:
We have this custom logic in our retry policy implementation that makes checking for cancellation token validity prior to entering the retry block a bit cumbersome to add; i.e. we'd need to add a similar translation of exception in order to preserve the behavior.
https://github.com/Azure/azure-iot-sdk-csharp/blob/480a27ba6a5a4486d2514b48cfefd5d42c30a6c1/iothub/device/src/TransientFaultHandling/RetryPolicy.cs#L281-L294

@drwill-ms: I inclined to leave the code in `main` as it is and introduce this change in `previews/v2`. What do you think?

Reply:
I'm fine with leaving main mostly the same.